chef/cookbooks/ceph/recipes/mds.rb

```diff
@@ -0,0 +1,47 @@
+if node.roles.include?("ceph-mon")
+  include_recipe "ceph::default"
+  include_recipe "ceph::conf"
+else
+  include_recipe "ceph::keyring"
+end
+include_recipe "ceph::server"
+
+execute "create mds keyring" do
+  command "ceph auth get-or-create mds.#{node["hostname"]} osd 'allow rwx' mds 'allow' mon 'allow profile mds' -o /etc/ceph/mds.#{node["hostname"]}.keyring ; chown ceph.ceph /etc/ceph/mds.#{node["hostname"]}.keyring"
+  not_if { File.exist?("/etc/ceph/mds.#{node["hostname"]}.keyring") }
+end
+
+service "ceph_mds" do
+  service_name "ceph-mds@#{node["hostname"]}"
+  supports restart: true, status: true
+  action [:enable, :start]
+  subscribes :restart, resources(template: "/etc/ceph/ceph.conf")
+end
+
+service "ceph-mds.target" do
+  action :enable
+end
+service "ceph.target" do
+  action :enable
+end
+
+# TODO: Not sure if the below will be racey or otherwise have timing issues
+# with MDS starting up or pool creation.
+
+execute "create data pool" do
+  # TODO: set sensible pg num
+  command "ceph osd pool create #{node[:ceph][:cephfs][:data_pool]} 256"
```

Comment:
Ah ok, if the default is set I guess just invoking ceph osd pool create should pick that up by itself, iirc